pkg/k8s/controllers/telemetryservice_controller.go

```diff
@@ -68,3 +94,78 @@ func (r *TelemetryServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&v1alpha1.TelemetryService{}).
 		Complete(r)
 }
+
+func CreateServiceIfNotExists(ctx context.Context, r *TelemetryServiceReconciler, ts *v1alpha1.TelemetryService, req ctrl.Request) error {
+	rlog := log.FromContext(ctx)
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tsNamespacedName.Name,
+			Namespace: tsNamespacedName.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{"app": tsNamespacedName.Name},
+			Ports: []corev1.ServicePort{
+				{
+					Port:       80,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.IntOrString{IntVal: 8080},
+				},
+			},
+			Type: corev1.ServiceTypeLoadBalancer,
+		},
+	}
+	rlog.Info("Start Creating service")
+	if err := r.Create(ctx, svc); err != nil {
+		rlog.Error(err, "Failed to create new Service", "Service.Namespace", svc.Namespace, "Service.Name", svc.Name)
+		return err
+	}
+	return nil
+}
+
+func CreateDeploymentIfNotExists(ctx context.Context, r *TelemetryServiceReconciler, ts *v1alpha1.TelemetryService, req ctrl.Request) error {
+	rlog := log.FromContext(ctx)
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tsNamespacedName.Name,
+			Namespace: tsNamespacedName.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": tsNamespacedName.Name},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": tsNamespacedName.Namespace},
```

Comment:
why this label use telemetryservice's namespace?

Reply:
Same reason as below

Reply:
In fact, in `telemetryservice_install.yaml`, use `telemetryServiceName` instead of `namespace`.

Reply:
Sorry, I just understood what your comment, and this is a typo for me. Let me fix it.